sdk/storage/azure-storage-blob/V10_V12.md

```diff
@@ -0,0 +1,379 @@
+```java
+// specify token properties
+String identifier = "identifier";
+BlobSASPermission permissions = new BlobSASPermission()
+    .withRead(true)
+    .withCreate(true)
+    .withDelete(true)
+    .withWrite(true); // We can also choose BlobContainerSasPermission, BlobServiceSasQueryParameters based on the object level we want to grant access.
+OffsetDateTime startTime = OffsetDateTime.now().minusDays(1);
+OffsetDateTime expiryTime = OffsetDateTime.now().plusDays(1);
+SasIpRange ipRange = new SasIpRange()
+    .setIpMax("0.0.0.0")
+    .setIpMax("255.255.255.255");
+SasProtocol sasProtocol = SasProtocol.HTTPS_HTTP;
+String cacheControl = "cache";
+String contentDisposition = "disposition";
+String contentEncoding = "encoding";
+String contentLanguage = "language";
+String contentType = "type";
+String version = BlobServiceVersion.V2019_02_02.getVersion();
+
+// build the token
+BlobServiceSasSignatureValues sasSignatureValues = new BlobServiceSasSignatureValues(version, sasProtocol, startTime, expiryTime, permissions.toString(), ipRange, identifier, cacheControl, contentDisposition, contentEncoding, contentLanguage, contentType);
+BlobServiceSasQueryParameters sasQueryParameters = sasSignatureValues.generateSasQueryParameters(new StorageSharedKeyCredential("<account-name>", "<account-key"));
+```
+
+## Build Client
+In V10/V11, core classes were built with constructors, taking a `URL` and an already constructed `HttpPipeline`.
+```java
+final BlockBlobURL blobURL = new BlockBlobURL(
+    new URL("https://" + "<account-name>" + ".blob.core.windows.net/mycontainer/myimage.jpg"),
+    StorageURL.createPipeline(new SharedKeyCredentials("<account-name>", "<account-key>"), new PipelineOptions()));
+```
+
+In V12, we have moved to a builder pattern. To replicate the above V10/V11 snippit:
+```java
+// To get the BlobClient, we can use builder to initialize.
+final BlobClient blobClient = new BlobClientBuilder()
+    .endpoint("https://" + "<account-name>" + ".blob.core.windows.net")
+    .containerName("mycontainer")
+    .blobName("myimage.jpg")
+    .credential(new StorageSharedKeyCredential("<account-name>", "<account-key>"))
+    .buildClient();
+// We can initialize BlockBlobClient from blobClient.
+final BlockBlobClient blockBlobClient = blobClient.getBlockBlobClient(); // we can also choose PageBlobClient, AppendBlobClient, and BlobClient
+```
+These builders can also build sync and async clients interchangably. Just select the correct "build" method. Additionally, if you already have the full URL, you can just specify that as the endpoint instead of naming container and blob individually, similar to V10/V11.
+```java
+final BlobAsyncClient blobAsyncClient = new BlobClientBuilder()
+    .endpoint("https://" + "<account-name>" + ".blob.core.windows.net")
```

Comment:
I think based on the description above this snippet, the endpoint is supposed to have the container and blob names already on it.

Reply:
Done.

Reply:
Done.